design/trackers.go

```diff
@@ -0,0 +1,125 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var trackerData = a.Type("TrackerData", func() {
+	a.Attribute("attributes", trackerAttributes)
+	a.Attribute("type", d.String, func() {
+		a.Enum("trackers")
+	})
+	a.Attribute("id", d.String, "ID of the tracker record", func() {
+		a.Example("42")
+	})
+	a.Attribute("links", genericLinks)
+	a.Required("type", "attributes")
+})
+
+var trackerLinks = a.Type("TrackerLinks", func() {
+	a.Attribute("self", d.String, func() {
+		a.Example("http://api.almighty.io/api/trackers/42")
+	})
+	a.Required("self")
+})
+
+// Defines a Media Type for Single Tracker Object
+var TrackerObject = JSONSingle("TrackerObject", "Single Tracker Payload", trackerData, trackerLinks)
```

Comment:
thanks, I wil add relationships.

I tried following naming convention but `JSONSingle` function appends `single` to the `name` value. So to achieve that I need to do -
```
var Tracker = JSONSingle("Tracker", "Single Tracker Pa....)
```
But in this case, this type will conflict with exisitng `tracker`.
Hence I had to do -
```
var TrackerObject = JSONSingle("Tracker", "Single Tracker Pa....)
```
So, in design I will refer it as `trackerObject` and elsewhere I will refer as `TrackerSingle`, But I did not find this correct way of doing it.